### PR TITLE
Support for Uprobes

### PR DESCRIPTION
**Pull request by jcanseco**

### CODE REVIEW

File: src/bpfd.h [hunk 1]
```diff
@@ -35,11 +37,18 @@
...
+	if (!tok)				\
+		goto invalid_command;		\
+	if (!sscanf(tok, "%"SCNu64" ", &var))	\
+		goto invalid_command;
+
 #define PARSE_ULL(var)				\
 	tok = strtok(NULL, " ");		\
 	if (!tok)				\
 		goto invalid_command;		\
-	if (!sscanf(tok, "%llu ", &var))		\
```
Comment (joelagnel) [inline src/bpfd.h:42]: remove extra tab

**Review by joelagnel — COMMENTED** [review]
Looks good other than the minor comment. Can you squash the 2 patches together?

File: src/bpfd.c [hunk 2]
```diff
@@ -483,6 +483,24 @@ int main(int argc, char **argv)
...
+			uint64_t offset;
+
+			PARSE_FIRST_INT(prog_fd);
+			PARSE_INT(type);
+			PARSE_STR(ev_name);
+			PARSE_STR(binary_path);
+			PARSE_UINT64(offset);
+			PARSE_INT(pid);
+
+			if (!bpf_attach_uprobe(prog_fd, type, ev_name, binary_path, offset, pid, NULL, NULL))
```
Comment (joelagnel) [inline src/bpfd.c]: I think we're leaking a perf reader here too, add a TODO comment about that.

File: src/bpfd.h (hunk 1 above)
Comment (jcanseco) [reply src/bpfd.h:42]: Do you mean to put back the extra tab? I removed it in order to align the "\\", but I can undo that if that's what you mean.

Comment (joelagnel) [reply src/bpfd.h:42]: yep, its superflous

Comment (joelagnel) [reply src/bpfd.h:42]: Oh ok I got confused. Leave the change in, its fine. Could you squash everything though, "uprobes support" can be one commit.

**Review by joelagnel — CHANGES_REQUESTED** [review]

**Comment by jcanseco** [comment]
I made all the requested changes and squashed all commits into one. PTAL and feel free to merge if it looks good :)

File: src/bpfd.h (hunk 1 above)
Comment (jcanseco) [reply src/bpfd.h:42]: Done.

**Review by joelagnel — APPROVED** [review]

**Pull request merged**